unskript-ctl/templates/template_script.j2

```diff
@@ -28,69 +34,81 @@ def _run_function(fn):
         output = output_buffer.getvalue()
     else:
         try:
-            fn = globals().get(fn)
+            fn = globals().get(fname)
             if _logger:
-                # Since lego and lego_printer are nested within the check function
-                # we can use co_consts to extract the lego_printer and strip off the 
-                # _printer to get the lego function name
+                # Since we are using decorator for controlling the timeout, the earlier way of getting const.co_name
+                # would not work because, the decorator mapping mangles the function name with the wrapper. 
+                # We rely on the calling function to get the check name as a mapping (dictionary).
                 chk_name = ''
-                for const in fn.__code__.co_consts:
-                    if isinstance(const, type(fn.__code__)):
-                        if const.co_name.endswith('printer'):
-                            chk_name = const.co_name.replace('_printer','')
+                if _script_to_check_mapping:
+                    chk_name = _script_to_check_mapping.get(fname)
                 _logger.debug(f"Starting to execute check {fn} {chk_name}")
-            fn() 
+            
+            poll(globals().get(fname), 
+                            step=1, 
+                            timeout={{ execution_timeout }},
+                            poll_forever=False,
+                            check_success= lambda v: v is not None)
             success = True
+        except polling2.TimeoutException:
+            # Polling timeout
+            if _logger:
+                _logger.debug(f"Execution completed for {fn}")
```

Comment:
While `{fn}` is useful, it's better to include the name of the check that is misbehaving.